File: frontend/lib/models/order.dart
import '../providers/cart.dart';
// import 'cart.dart';

class Order {
  final String id;
  final String invoiceNumber;
  final String userId;
  final double amount;
  final List<CartItem> items;
  final String datetime;

  const Order(
      {required this.id,
      required this.invoiceNumber,
      required this.userId,
      required this.amount,
      required this.items,
      required this.datetime});

  factory Order.fromJson(Map<String, dynamic> json) {
    return Order(
        id: json['id'],
        invoiceNumber: json['invoicenumber'],
        userId: json['userid'],
        amount: json['amount'],
        items: CartItem.fromListJson(json['items']),
        datetime: json['datetime']);
  }
  static List<Order> fromListJson(List<dynamic> json) {
    List<Order> result = <Order>[];
    for (Map<String, dynamic> d in json) {
      result.add(Order.fromJson(d));
    }
    return result;
  }
}

File: aspnet/Model/order.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
namespace aspnet.models;
public class Order
{
    [BsonId][BsonRepresentation(BsonType.ObjectId)] 
    public string? id { get; set; }
    [BsonElement]
    public string invoicenumber { get; set; }
    [BsonElement]
    public string userid { get; set; }
    [BsonElement]
    public double amount { get; set; }
    [BsonElement]
    public List<CartItem> items { get; set; }
    [BsonElement]
    public string datetime { get; set; }

    public Order(string Id, string Invoicenumber, string Userid, double Amount, List<CartItem> Items, string Datetime)
    {
        this.id = Id;
        this.invoicenumber = Invoicenumber;
        this.userid = Userid;
        this.amount = Amount;
        this.items = Items;
        this.datetime = Datetime;
    }
}


